### PR TITLE
Sp2285 2

### PR DESCRIPTION
**Pull request by CeltonMcGrath**

### CODE REVIEW

File: src/context/spt_js/client_api.js [hunk 1]
```diff
@@ -1344,7 +1344,20 @@ TacticServerStub = function() {
...
+        }
+        var on_error = function(e) {
+            callback(e);
+        };
 
+        this._delegate("ping", arguments, kwargs, null, callback, on_error);
+        return;
+    }
+    
+
```
Comment (CeltonMcGrath) [inline src/context/spt_js/client_api.js]: I am not sure why the callback parameter is 'cbjs_action' or 'callback', it should be renamed 'on_error_callback' . Do you agree?

File: src/pyasm/biz/file.py [hunk 2]
```diff
@@ -638,6 +638,16 @@ def _process_pdf(my, file_name):
         else:
             print "Warning: [%s] did not get created from pdf" % tmp_web_path
 
+            
+    def get_free_size(my):
+        '''When free_size project setting is true, an thumb is created
+        proportional to original size, but with transparent border.
+        Conversion is RGBA instead of default RBG for icon creation.'''
+        from pyasm.prod.biz import ProdSetting
```
Comment (CeltonMcGrath) [inline src/pyasm/biz/file.py]: This is incorrect... can you suggest a better docstring?

Comment (oboreo) [inline src/pyasm/biz/file.py]: from pyasm.biz import ProdSetting

File: src/pyasm/biz/file.py [hunk 3]
```diff
@@ -859,17 +873,23 @@ def _resize_image(my, large_path, small_path, thumb_size):
...
-                    # then paste to white image
-                    im2 = Image.new( "RGB", thumb_size, (255,255,255) )
-                    offset = (thumb_size[0]/2) - (im.size[0]/2)
+                    
+                    if free_size:                        
+                        im2 = Image.new( "RGBA", (base_width, base_height), (255,255,255,0) )
+                        offset = 0
+                    else:
+                        # then paste to white image
+                        im2 = Image.new( "RGB", thumb_size, (255,255,255,0) )
```
Comment (oboreo) [inline src/pyasm/biz/file.py]: (255,255,255)

File: src/tactic/ui/widget/embed_wdg.py [hunk 4]
```diff
@@ -100,24 +102,36 @@ def get_display(my):
...
+                    "source_types": source_types, 
+                    "poster": poster, 
+                    "preload": preload, 
+                    "controls": controls, 
+                    "width": width, 
+                    "height": height, 
+                    "index": index
+            }
+                    
+            if my.kwargs.get("video_class") == "videojswdg":
```
Comment (oboreo) [inline src/tactic/ui/widget/embed_wdg.py]: use CamelCase, VideoJsWdg

File: src/tactic/ui/widget/embed_wdg.py [hunk 5]
```diff
@@ -66,7 +66,9 @@ def get_display(my):
         #div = DivWdg()
```
Comment (oboreo) [inline src/tactic/ui/widget/embed_wdg.py]: should define ARGS_KEYS =  {
'video_class': .....  refer to table_layout_wdg.py
}

**Pull request closed without merging**